addons/website/models/website.py

```diff
@@ -477,24 +483,123 @@ def set_features(selected_features):
 
             return pages_views
 
-        def configure_page(page_code, snippet_list, pages_views, cta_data):
+        dynamic_snippets = {
+            's_dynamic_snippet_products': {
+                'filter_name': 'website_sale.dynamic_filter_newest_products',
+                'template_key': 'website_sale.dynamic_filter_template_product_product_borderless_1',
+                'data_attributes': {'number-of-records': '16', 'product-category-id': 'all', 'carousel-interval': '5000'},
+            },
+            's_events': {
+                'filter_name': 'website_event.website_snippet_filter_event_list',
+                'template_key': 'website_event.dynamic_filter_template_event_event_picture',
+                'data_attributes': {'number-of-records': '4'},
+            },
+            's_blog_posts': {
+                'filter_name': 'website_blog.dynamic_filter_latest_blog_posts',
+                'template_key': 'website_blog.dynamic_filter_template_blog_post_big_picture',
+                'data_attributes': {'number-of-records': '16', 'filter-by-blog-id': '-1', 'number-of-elements': '3'},
+                'hover_effect': 's_blog_posts_effect_marley'
+            },
```

Comment:
@sobo-odoo I am sure inventing this new syntax is the result of back and forth in testing and review but when the technical specs differ that much from the original ones... try contacting @rdeodoo or me :wink: 

Pretty sure a new syntax and that much code to support the dynamic snippet only, this is already a possibility without this. Well at least it will be by just implementing the original specification: just authorizing to enter the whole xmlid of a snippet (module included) in the manifest instead of supposing it is a website one. Once you have done that, if a theme wants to add customizations, he can do customization by creating a new snippet with primary="True" and xpath (just like normal snippet customizations), in XML.

Although, it's true that for a website_sale/website_blog snippet, this will require to create sub-themes like we had before, like theme_orchid_sale. It's annoying but it will be the right way. I am pretty sure we'll want that sooner or later with page templates that BSO is making anyway. In your PR, you also mention event/blog snippets in the website application, this is not right :wink: Maybe we should discuss this with @rdeodoo first (so wait before changing) but I think he'll agree :wink:

Reply:
Indeed, the first thing I did was to add the possibility to add other modules snippets in the `snippet_lists`, but when I tested, I saw that the dynamic snippets were dropped but they were not visible because they were not initialized (so they had no filter and template,...). 
So I tried to create a new snippet in the themes, in order to xpath to add the different attributes like the other theme snippets (just to see if it was possible) but the issue is that it couldn't find the template id no matter what I put (it didn't work with `inherit_id="website_sale.s_dynamic_snippet_products"` nor with `s_dynamic_snippet_products`) so I guessed that these ids were "read" before the modules installation. And so it could not work when installing Website for the first time with no other modules installed before. (I tried with one of the dynamic snippet modules installed before installing Website and in that case it worked) :grimacing:

I just tested with the `primary="True"` you just mentioned (maybe that is what I missed) but it still doesn't seem to be working (maybe I am doing something wrong... :sweat_smile:)

So I figured that I needed to find an other way to initialize/customize them easily, which ended in this PR. But I also don't like it (that's why I asked to several people if it was not too "monstrueux" to do it like that before)
So for me it was not really a spec change but a way to make the base case work (if we don't see the snippet then it doesn't work :grimacing: )

(Just telling so you know how it ended like this when we will discuss it :slightly_smiling_face:)

Reply:
I agree that there might be some code to consider overkill here (we will need to fix & maintain it).
But it might be easier to discuss this one IRL to see spec-wise what we really want and see if we can get rid of it?
We can do that this afternoon when QSM is there :+1: